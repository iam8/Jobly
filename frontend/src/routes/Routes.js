import React from "react";
import { Route, Switch, Redirect } from "react-router-dom";

import Home from "../homepage/Home";
import LoginForm from "../auth/LoginForm";
import SignupForm from "../auth/SignupForm";
import Companies from "../companies/Companies";
import Company from "../companies/Company";
import Jobs from "../jobs/Jobs";
import EditProfileForm from "../profiles/EditProfileForm";
import ProtectedRoute from "./ProtectedRoute";


/**
 * All routes for Jobly website.
 *
 * Props:
 *  - login(): log in an existing user; passed by parent
 *  - signup(): register a new user; passed by parent
 *
 * Route list:
 *  - /companies/:handle
 *  - /companies
 *  - /jobs
 *  - /login
 *  - /signup
 *  - /profile
 *  - /
 *
 * Non-matching URLs will redirect to / (home).
 **/
function Routes({login, signup}) {

    return (
        <Switch>
            <Route exact path="/companies/:handle">
                <ProtectedRoute>
                    <Company />
                </ProtectedRoute>
            </Route>

            <Route exact path="/companies">
                <ProtectedRoute>
                    <Companies />
                </ProtectedRoute>
            </Route>

            <Route exact path="/jobs">
                <ProtectedRoute>
                    <Jobs />
                </ProtectedRoute>
            </Route>

            <Route exact path="/login">
                <LoginForm login={login}/>
            </Route>

            <Route exact path="/signup">
                <SignupForm signup={signup}/>
            </Route>

            <Route exact path="/profile">
                <ProtectedRoute>
                    <EditProfileForm />
                </ProtectedRoute>
            </Route>

            <Route exact path="/">
                <Home />
            </Route>

            <Redirect to="/" />
        </Switch>
    )
}


export default Routes;
