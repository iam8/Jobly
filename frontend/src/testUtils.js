import React from "react";
import UserContext from "./auth/UserContext";


const demoUser = {
    username: "testuser",
    firstName: "testfirst",
    lastName: "testlast",
    email: "test@test.net",
};


const DemoUserProvider = ({ children, currentUser = demoUser, hasAppliedToJob = () => false }) => (
    <UserContext.Provider value={{ currentUser, hasAppliedToJob }}>
        {children}
    </UserContext.Provider>
);


export { DemoUserProvider };
