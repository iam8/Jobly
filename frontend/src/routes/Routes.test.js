import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import { DemoUserProvider as UserProvider } from "../testUtils";
import Routes from "./Routes";


test("Renders without crashing", () => {
    render (
        <MemoryRouter>
            <UserProvider>
                <Routes />
            </UserProvider>
        </MemoryRouter>
    );
})


test("Matches snapshot", () => {
    const { asFragment } = render(
        <MemoryRouter>
            <UserProvider>
                <Routes />
            </UserProvider>
        </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
});