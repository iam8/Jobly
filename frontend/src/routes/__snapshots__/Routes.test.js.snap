// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Home"
  >
    <h1>
      Jobly
    </h1>
    <h3>
      Apply for jobs from one spot, for free
    </h3>
    <h2
      class="Home-welcome"
    >
      Welcome, testfirst!
    </h2>
  </div>
</DocumentFragment>
`;
